File: app/src/main/java/com/alexkn/syntact/data/common/AppDatabase.java
package com.alexkn.syntact.data.common;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.alexkn.syntact.data.dao.BucketDao;
import com.alexkn.syntact.data.dao.ClueDao;
import com.alexkn.syntact.data.dao.LetterDao;
import com.alexkn.syntact.data.dao.SolvableItemDao;
import com.alexkn.syntact.data.util.Converters;
import com.alexkn.syntact.data.model.Bucket;
import com.alexkn.syntact.data.model.Clue;
import com.alexkn.syntact.data.model.Letter;
import com.alexkn.syntact.data.model.SolvableItem;
import com.alexkn.syntact.data.model.views.BucketDetail;

@Database(entities = {SolvableItem.class, Bucket.class, Letter.class, Clue.class},views = {BucketDetail.class},
        version = 30)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {

    private static volatile AppDatabase instance;

    public abstract SolvableItemDao solvableItemDao();

    public abstract BucketDao bucketDao();

    public abstract LetterDao letterDao();

    public abstract ClueDao clueDao();

    public static AppDatabase getDatabase(final Context context) {

        if (instance == null) {
            synchronized (AppDatabase.class) {
                if (instance == null) {
                    instance = Room
                            .databaseBuilder(context.getApplicationContext(), AppDatabase.class,
                                    "app_database").fallbackToDestructiveMigration().build();
                }
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/alexkn/syntact/data/common/AppDatabase.kt
package com.alexkn.syntact.data.common

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.alexkn.syntact.core.model.Clue
import com.alexkn.syntact.core.model.Deck
import com.alexkn.syntact.core.model.Preferences
import com.alexkn.syntact.core.model.SolvableItem
import com.alexkn.syntact.data.common.util.Converters
import com.alexkn.syntact.data.dao.ClueDao
import com.alexkn.syntact.data.dao.DeckDao
import com.alexkn.syntact.data.dao.PreferencesDao
import com.alexkn.syntact.data.dao.SolvableItemDao

@Database(
        entities = [SolvableItem::class, Deck::class, Clue::class, Preferences::class],
        version = 73)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun solvableItemDao(): SolvableItemDao

    abstract fun deckDao(): DeckDao

    abstract fun clueDao(): ClueDao

    abstract fun preferencesDao(): PreferencesDao

}

